sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -735,7 +737,7 @@ def _rename_path(self, rename_source: str, **kwargs: Any) -> Dict[str, Any]:
         except HttpResponseError as error:
             process_storage_error(error)
 
-    def _get_path_properties(self, **kwargs: Any) -> Union["DirectoryProperties", "FileProperties"]:
+    def _get_path_properties(self, **kwargs: Any) -> "BlobProperties":
```

Comment:
I think we talked about this before. This cannot be `BlobProperties` and must stay as `Union["DirectoryProperties", "FileProperties"]`. The `cls` from the file and directory clients convert the `BlobProperties` to `FileProperties` or `DirectoryProperties` respectively. You can use an ignore for this one or try casting to the `Union`.

Reply:
Sorry I forgot to make the change! I've updated this for both sync and async versions :)